Panic when setting proxy config

```
asaoines@validator-0:~/celo-validator-node$ sudo docker stop celo-validator && sudo docker rm celo-validator && sudo docker run --name celo-validator -it --restart always -p 30303:30303 -p 30303:30303/udp -v $PWD:/root/.celo us.gcr.io/celo-testnet/celo-node:baklava --verbosity 4 --networkid 12219 --syncmode full --mine --istanbul.blockperiod=5 --istanbul.requesttimeout=3000 --etherbase 6729622cb0535f168e744c4ecbc380fb230cb9b2 --nodiscover --proxy.proxied --proxy.proxyenodeurlpair=enode://7571be2ca67baa99930f8a7655dd999875654c6960baee2d0657e80e43c2ebc65ba7aedb8127c8ff52c12281e4f2bf5685c61a604cbf28ed943edfd8675f5b47@34.65.75.14:30503;enode://7571be2ca67baa99930f8a7655dd999875654c6960baee2d0657e80e43c2ebc65ba7aedb8127c8ff52c12281e4f2bf5685c61a604cbf28ed943edfd8675f5b47@34.65.75.14:30303 --unlock=6729622cb0535f168e744c4ecbc380fb230cb9b2 --password /root/.celo/.password --ethstats=asa@baklava-ethstats.celo-testnet.org
celo-validator
celo-validator
DEBUG[12-08|17:24:45.139] Sanitizing Go's GC trigger               percent=100
INFO [12-08|17:24:45.139] Maximum peer count                       ETH=25 LES=99 total=124
panic: runtime error: index out of range

goroutine 1 [running]:
github.com/ethereum/go-ethereum/cmd/utils.SetProxyConfig(0xc0000fa160, 0xc000097a68, 0xc000097800)
        /go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/cmd/utils/flags.go:1279 +0xbc5
main.makeConfigNode(0xc0000fa160, 0xc000097100, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/cmd/geth/config.go:127 +0x3af
main.makeFullNode(0xc0000fa160, 0x0)
        /go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/cmd/geth/config.go:154 +0x7e
main.geth(0xc0000fa160, 0x0, 0x0)
        /go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/cmd/geth/main.go:280 +0xc9
github.com/ethereum/go-ethereum/vendor/gopkg.in/urfave/cli%2ev1.HandleAction(0x10dcd80, 0x1305db0, 0xc0000fa160, 0xc0000fa160, 0xc00051ff58)
        /go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/vendor/gopkg.in/urfave/cli.v1/app.go:490 +0xc8
github.com/ethereum/go-ethereum/vendor/gopkg.in/urfave/cli%2ev1.(*App).Run(0xc0000b64e0, 0xc0000ac000, 0xf, 0xf, 0x0, 0x0)
        /go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/vendor/gopkg.in/urfave/cli.v1/app.go:264 +0x59d
main.main()
        /go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/cmd/geth/main.go:267 +0x55
-bash: enode://7571be2ca67baa99930f8a7655dd999875654c6960baee2d0657e80e43c2ebc65ba7aedb8127c8ff52c12281e4f2bf5685c61a604cbf28ed943edfd8675f5b47@34.65.75.14:30303: No such file or directory
```

--- a/cmd/utils/flags.go
+++ b/cmd/utils/flags.go
@@ -1271,6 +1271,9 @@ func SetProxyConfig(ctx *cli.Context, nodeCfg *node.Config, ethCfg *eth.Config) 
 			Fatalf("Option --%s must be used if option --%s is used", ProxyEnodeURLPairFlag.Name, ProxiedFlag.Name)
 		} else {
 			proxyEnodeURLPair := strings.Split(ctx.String(ProxyEnodeURLPairFlag.Name), ";")
+			if len(proxyEnodeURLPair) != 2 {
+				Fatalf("Invalid usage for option --%s", ProxyEnodeURLPairFlag.Name)
+			}
 
 			var err error
 			if ethCfg.Istanbul.ProxyInternalFacingNode, err = enode.ParseV4(proxyEnodeURLPair[0]); err != nil {

--- a/consensus/consensus.go
+++ b/consensus/consensus.go
@@ -151,6 +151,9 @@ type Handler interface {
 
 	// UnregisterPeer will notify the consensus engine that a new peer has been removed
 	UnregisterPeer(peer Peer, fromProxiedNode bool)
+
+	// ConnectToVals
+	ConnectToVals()
 }
 
 // PoW is a consensus engine based on proof-of-work.

--- a/consensus/istanbul/backend/backend.go
+++ b/consensus/istanbul/backend/backend.go
@@ -712,3 +712,13 @@ func (sb *Backend) ValidatorAddress() common.Address {
 	}
 	return localAddress
 }
+
+func (sb *Backend) ConnectToVals() {
+	// If this is a proxy, then refresh the val peers.  Note that this will be done within Backend.Start
+	// for non proxied validators
+	if sb.config.Proxy {
+		headBlock := sb.GetCurrentHeadBlock()
+		valset := sb.getValidators(headBlock.Number().Uint64(), headBlock.Hash())
+		sb.RefreshValPeers(valset)
+	}
+}

--- a/consensus/istanbul/backend/engine.go
+++ b/consensus/istanbul/backend/engine.go
@@ -658,6 +658,10 @@ func (sb *Backend) Start(hasBadBlock func(common.Hash) bool,
 		}
 
 		go sb.sendValEnodesShareMsgs()
+	} else {
+		headBlock := sb.GetCurrentHeadBlock()
+		valset := sb.getValidators(headBlock.Number().Uint64(), headBlock.Hash())
+		sb.RefreshValPeers(valset)
 	}
 
 	return nil

--- a/eth/handler.go
+++ b/eth/handler.go
@@ -245,6 +245,11 @@ func (pm *ProtocolManager) Start(maxPeers int) {
 	// start sync handlers
 	go pm.syncer()
 	go pm.txsyncLoop()
+
+	// Reconnect all the peer connections from the on-disk val enode table
+	if handler, ok := pm.engine.(consensus.Handler); ok {
+		handler.ConnectToVals()
+	}
 }
 
 func (pm *ProtocolManager) Stop() {

